system_tests/program_gas_test.go

```diff
@@ -23,6 +24,54 @@ import (
 	"github.com/offchainlabs/nitro/util/testhelpers"
 )
 
+func TestGasUsageOfHostiosThatDontHaveGoodEVMEquivalents(t *testing.T) {
+	builder := setupGasCostTest(t)
+	auth := builder.L2Info.GetDefaultTransactOpts("Owner", builder.ctx)
+	stylusProgram := deployWasm(t, builder.ctx, auth, builder.L2.Client, rustFile("hostio-test"))
+	matchSnake := regexp.MustCompile("_[a-z]")
+
+	for _, tc := range []struct {
+		hostio      string
+		expectedInk uint64
+	}{
+		{hostio: "read_args", expectedInk: 8400 + 5040},
+		{hostio: "write_result", expectedInk: 8400 + (16381+55*(10000-32))*2},
+		{hostio: "storage_cache_bytes32", expectedInk: 8400 + (13440-8400)*2},
+		{hostio: "msg_reentrant", expectedInk: 8400},
+		{hostio: "pay_for_memory_grow", expectedInk: 9320660000},
+	} {
+		t.Run(tc.hostio, func(t *testing.T) {
+			funcName := matchSnake.ReplaceAllStringFunc(tc.hostio, func(s string) string {
+				return strings.ToUpper(strings.TrimPrefix(s, "_"))
+			})
+			signature := fmt.Sprintf("%v()", funcName)
+			data := crypto.Keccak256([]byte(signature))[:4]
```

Comment:
You could create an interface in the solidity mocks to facilitate the generation of the call args.

Reply:
I want to avoid doing that 😬.
Ideally we could use an ABI that is generated directly from the rust smart contract, but we don't have a setup for this yet, so I decided to build calldata in a more low level approach since tests are simple.